api/tests/core/bookings/test_api.py

```diff
@@ -844,6 +844,39 @@ def test_raise_external_booking_error_if_trying_to_cancel_with_external_server_e
         assert booking.status is not BookingStatus.CANCELLED
         assert not booking.cancellationReason
 
+    def test_raise_external_booking_error_if_trying_to_cancel_with_already_canceled_ticket(self, requests_mock):
+        external_url = "https://book_my_offer.com"
+        provider = providers_factories.ProviderFactory(
+            name="Technical provider",
+            bookingExternalUrl=external_url + "/book",
+            cancelExternalUrl=external_url + "/cancel",
+        )
+        providers_factories.OffererProviderFactory(provider=provider)
+        stock = offers_factories.EventStockFactory(
+            lastProvider=provider,
+            offer__subcategoryId=subcategories.SEANCE_ESSAI_PRATIQUE_ART.id,
+            offer__lastProvider=provider,
+            idAtProviders="",
+        )
+        booking = bookings_factories.BookingFactory(stock=stock)
+        offerer_factories.ApiKeyFactory(offerer=booking.offerer)
+        external_bookings_factories.ExternalBookingFactory(booking=booking, barcode="1234567890123")
+        requests_mock.post(
+            external_url + "/cancel",
+            json={"error": "already_cancelled"},
+            status_code=409,
+        )
+
+        try:
+            api.cancel_booking_by_beneficiary(booking.user, booking)
+        except Exception:  # pylint: disable=broad-except
+            # Should not raise an exception
+            # if the following line is executed the code is broken
+            pytest.fail("Should not raise an exception")
```

Comment:
normalement la syntaxe pour tester des exceptions est `with pytest.raises(Exception):` mais pour tester qu'une fonction ne renvoit pas d'erreur tu peux écrire `assert api.cancel_booking_by_beneficiary(booking.user, booking) is None`